iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -246,42 +243,63 @@ bool ProcessManager::requestShutdownOfProcess(const RouDiProcess& process,
                                       (shutdownPolicy == ShutdownPolicy::SIG_KILL ? SIGKILL : SIGTERM));
+
     if (killC.hasErrors())
     {
-        if (shutdownLog == ShutdownLog::FULL)
+        evaluateKillError(process, killC, shutdownPolicy);
+        return false;
+    }
+    return true;
+}
+
+bool ProcessManager::isProcessAlive(const RouDiProcess& process) noexcept
+{
+    static constexpr int32_t ERROR_CODE = -1;
+    auto checkCommand = iox::cxx::makeSmartC(kill,
+                                             iox::cxx::ReturnMode::PRE_DEFINED_ERROR_CODE,
+                                             {ERROR_CODE},
+                                             {ESRCH},
+                                             static_cast<pid_t>(process.getPid()),
+                                             SIGTERM);
+
+    if (checkCommand.getErrNum() == ESRCH)
+    {
+        return false;
+    }
+    else
+    {
+        if (checkCommand.hasErrors())
         {
-            switch (killC.getErrNum())
-            {
-            case EINVAL:
-                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
-                          << "' could not be killed with "
-                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
-                          << ", because the signal sent was invalid.";
-                break;
-            case EPERM:
-                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
-                          << "' could not be killed with "
-                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
-                          << ", because RouDi doesn't have the permission to send the signal to the target processes.";
-                break;
-            case ESRCH:
-                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
-                          << "' could not be killed with "
-                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
-                          << ", because the target process or process group does not exist.";
-                break;
-            default:
-                LogWarn() << "Process ID " << process.getPid() << " named '" << process.getName()
-                          << "' could not be killed with"
-                          << (shutdownPolicy == ShutdownPolicy::SIG_KILL ? "SIGKILL" : "SIGTERM")
-                          << " for unknown reason: ’" << killC.getErrorString() << "'";
-            }
+            evaluateKillError(process, checkCommand, ShutdownPolicy::SIG_TERM);
         }
-        return false;
+        return true;
     }
     return true;
 }
 
+void ProcessManager::evaluateKillError(const RouDiProcess& process,
+                                       const iox::cxx::SmartC<int(pid_t __pid, int __sig), int32_t, pid_t, int> cmd,
```

Comment:
please create an alias for this and pass it by reference or if feasible, just pass the errnum and error string